### PR TITLE
Only publish MQTT on LoRa Tx if packet is from us and on Rx if not

### PR DESCRIPTION
**Pull request by GUVWAF**
Currently we're only publishing to MQTT when we're *transmitting* a packet via LoRa. Nevertheless, we still see *received* packets on MQTT as well, but this is only because we forced a node to always rebroadcast when MQTT is enabled:
https://github.com/meshtastic/firmware/blob/143ee9cdf61773f0d719a9162affd96427f8b3ad/src/mesh/FloodingRouter.cpp#L24 

However, as observed in #3138, direct messages towards the MQTT node will not be published, because we don't rebroadcast if we're the destination.

With this, we're publishing when transmitting via LoRa only when we're the original transmitter, and we publish when receiving when we're not the original transmitter. This way we can get rid of the "hack" where we're always rebroadcasting via LoRa when MQTT is enabled, and we'll see direct messages to the MQTT node as well. 

Since we already don't publish packets to MQTT if they came via MQTT, this will not create any extra traffic when bridging networks.

Also fixed a possible issue where `env->packet` contains the encrypted version of the packet and we're trying to print `decoded.portnum`.

### CODE REVIEW

File: src/mesh/Router.cpp [hunk 1]
```diff
@@ -257,10 +257,9 @@ ErrorCode Router::send(meshtastic_MeshPacket *p)
             return encodeResult; // FIXME - this isn't a valid ErrorCode
         }
 
-        if (moduleConfig.mqtt.enabled) {
-            LOG_INFO("Should encrypt MQTT?: %d\n", moduleConfig.mqtt.encryption_enabled);
-            if (mqtt)
-                mqtt->onSend(*p, *p_decoded, chIndex);
+        // Only publish to MQTT if we're the original transmitter of the packet
+        if (moduleConfig.mqtt.enabled && p->from == nodeDB.getNodeNum() && mqtt) {
```
Comment (thebentern) [inline src/mesh/Router.cpp:261]: Just confirming, we would still "rebroadcast" a message to MQTT received from someone else transmitting on lora?

Comment (GUVWAF) [reply src/mesh/Router.cpp:261]: Yes, in this case it's handled upon receiving the packet via LoRa: 
https://github.com/meshtastic/firmware/blob/834490175d312dd7c4a2d6049185f030fffe85a3/src/mesh/Router.cpp#L454-L456

**Comment by GUVWAF** [comment]
Converting to draft for now. I just realized we should filter received packets with `wasSeenRecently()` to avoid getting duplicates. Need to find a clean way to do this.

**Comment by GUVWAF** [comment]
> I just realized we should filter received packets with `wasSeenRecently()` to avoid getting duplicates.

After further testing I found that this is not needed because it is already done in `shouldFilterReceived()` which is called in `Router::perhapsHandleReceived()` before `handleReceived()`, where we send to MQTT. 

This should be good to go then :)

**Pull request merged**